components/FlexibleHeader/BUILD

```diff
@@ -48,12 +48,27 @@ mdc_objc_library(
     visibility = ["//visibility:public"],
 )
 
+mdc_objc_library(
+    name = "CanAlwaysExpandToMaximumHeight",
+    srcs = native.glob(["src/CanAlwaysExpandToMaximumHeight/*.m"]),
```

Comment:
You do not have to have the files in a separate folder in order to exclude them from glob'ed sources.

Reply:
https://docs.bazel.build/versions/master/skylark/lib/native.html#glob

Reply:
Am aware, but this is the pattern we've followed for themers as well. I don't feel that we should break from that pattern.

Reply:
If our reasoning is to prevent internal clients from using this broadly, we can handle that internally. I'm not sure I understand why it warrants separate modules in Cocoapods and Bazel. Given that the size of the library seems pretty small (it's just code), can we use the glob excludes/includes in our internal configs to separate it there?

Reply:
If we feel that separate modules would be better maintained using glob exclusion patterns then we should change that pattern across all of our extension targets.

I do not feel that maintaining a set of glob exclusions (especially if/when we add a few more opt-in behaviors in the future) is a scalable or desirable engineering solution here.

Reply:
Agreed that it may be challenging to maintain long-term, but the justification for this being separate *at all* was that internally we want to limit its use. The number of instances where we want internal clients to deviate significantly in usage from the public APIs should be exceedingly low, so I hope we won't need to scale this to a large number of components.  Given that it's a peculiarity of our desires for internal usage, I don't like that we're restructuring MDC-iOS for everyone else.  It doesn't seem to provide additional value for external clients and the internal solution is pretty limited and very achievable.

Reply:
That's a fair argument. It raises the good question of how we might encourage limited use of such APIs for external clients.

Reply:
@romoore @randallli If you feel that the extensions pattern needs to be called into question I'd prefer if we can not block on this client-requested feature and pursue that discussion on a separate thread.

Reply:
I'm not calling into question our extensions approach to code. My understanding is that the "extensions pattern" was intended to reduce dependencies where clients might not desire them and keep our libraries/frameworks small. I agree with the stated goal and implementation of our extensions. For example, not pulling in all of Palettes just to get an AppBar.  In this case, I don't see that adding a couple of source files will significantly increase build targets for clients. I could be wrong, but I didn't see any new dependencies for the additional code.

What I **am** questioning is the justification for separating this out of the base component at all.

> We intend to keep usage of this API fairly restricted internally, so we have added the behavior as a separate target that we'll whitelist internally to clients. The API is still supported by the typical API contract for public clients.

The stated goal "keep usage of this API fairly restricted internally" can be achieved through internal mechanisms. My opposition is to unnecessary external structures to support internal requirements that don't necessitate that structure.